lua/entities/gmod_wire_expression2/core/custom/admincore.lua

```diff
@@ -56,6 +56,16 @@ e2function void explosion(vector pos,number mag)
 	explode:Fire("Explode",0,0)
 end
 
+e2function void explosion(vector pos,number mag,entity attacker)
+	if not self.player:IsAdmin() and allowexplosion != true then return end
+	local explode=ents.Create("env_explosion")
+	explode:SetPos(Vector(pos[1],pos[2],pos[3]))
+	explode:SetOwner(attacker)
+	explode:Spawn()
+	explode:SetKeyValue("iMagnitude",mag)
+	explode:Fire("Explode",0,0)
+end

```

Comment:
Is `allowexplosion` defined somewhere?

```suggestion
e2function void explosion(vector pos, number mag, entity attacker)
	if not self.player:IsAdmin() and not allowexplosion then return end

	local explode = ents.Create( "env_explosion" )
	explode:SetPos( Vector( pos[1], pos[2], pos[3] ) )
	explode:SetOwner( attacker )
	explode:Spawn()
	explode:SetKeyValue( "iMagnitude", mag )
	explode:Fire( "Explode", 0, 0 )
end

```

Reply:
allowexplosion is defined by a concmd, lets explosion stuff be enabled/disabled